flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/internals/KinesisDataFetcher.java

```diff
@@ -458,7 +458,7 @@ public void advanceLastDiscoveredShardOfStream(String stream, String shardId) {
 	 * 3. Update the subscribedStreamsToLastDiscoveredShardIds state so that we won't get shards
 	 *    that we have already seen before the next time this function is called
 	 */
-	private List<KinesisStreamShard> discoverNewShardsToSubscribe() throws InterruptedException {
+	public List<KinesisStreamShard> discoverNewShardsToSubscribe() throws InterruptedException {
```

Comment:
We would probably need a big refactor of the Kinesis code to be able to not expose this method, by separating concerns of partition discovery and record fetching. The exposure is not nice, but I think we have to do it for now ...